makefile

```diff
@@ -73,7 +73,7 @@ bgm13:
 	cd builds; rm -rf $@; mkdir $@; cd $@; cmake -DDEVICE=BGM13P22F512GA ../..; make;
 
 bgm1:
-	cd builds; rm -rf $@; mkdir $@; cd $@; cmake -DDEVICE=BGM111A256V2 -DBOARD=BGM11_BRD4300A ../..; make;
+	cd builds; rm -rf $@; mkdir $@; cd $@; cmake -DDEVICE=BGM111A256V2 -DBOARD=BGM11_BRD4300A ../..; make -j4;
```

Comment:
Now that I think about this maybe `make -j$(nproc)` makes more sense? Maybe we could look at the OS supported Darwin or Linux and offer the maximum number of cores available, i.e. `'sysctl -n hw.ncpu` for mac?

Reply:
Mainly because this looks like it might cause issues for anyone compiling on just 2 cores

Reply:
Looks like this is the first OS specific check we'll be adding in that case. Should that be a problem?

Reply:
@anujdeshpande right now this build system only support linux and MacOS (as far as I am aware). I don't think it's an issue to do a quick check before running on available cores.